pkg/segment/query/metricsquery.go

```diff
@@ -461,29 +478,53 @@ func getRegexMatchedMetricNames(mSegSearchReq *structs.MetricsSearchRequest, reg
 }
 
 func GetSeriesCardinalityOverTimeRange(timeRange *dtu.MetricsTimeRange, myid int64) (uint64, error) {
-	querySummary := summary.InitQuerySummary(summary.METRICS, rutils.GetNextQid())
-	defer querySummary.LogMetricsQuerySummary(myid)
-	tagsTreeReaders, err := GetAllTagsTreesWithinTimeRange(timeRange, myid, querySummary)
+	diff := uint64(timeRange.EndEpochSec - timeRange.StartEpochSec)
+	if diff <= 0 {
+		log.Errorf("GetSeriesCardinalityOverTimeRange: invalid time range, diff: %+v, start: %v, end: %v",
+			diff, timeRange.EndEpochSec, timeRange.StartEpochSec)
+		return 0, errors.New("Invalid time range")
+	}
+
+	pastXhours := max(1, diff/3600)
+	asCounts, err := usageStats.GetActiveSeriesCounts(pastXhours, myid)
 	if err != nil {
-		log.Errorf("GetSeriesCardinalityOverTimeRange: failed to get tags trees within time range %+v; err=%v", timeRange, err)
+		log.Errorf("GetSeriesCardinalityOverTimeRange: failed to get card, err: %v", err)
```

Comment:
nit: IMO in logs a little more verbosity is worth a little more clarity
```suggestion
		log.Errorf("GetSeriesCardinalityOverTimeRange: failed to get cardinality, err: %v", err)
```